config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ServerBridge.java

```diff
@@ -0,0 +1,274 @@
+    private static final String ClientConstants = "org.jboss.as.controller.client.helpers.ClientConstants";
+
+    private static final String createHostController = "createHostController";
+    private static final String createStandaloneServer = "createStandaloneServer";
+    private static final String start = "start";
+    private static final String getModelControllerClient = "getModelControllerClient";
+    private static final String execute = "execute";
+    private static final String fromJSONString = "fromJSONString";
+    private static final String get = "get";
+    private static final String set = "set";
+    private static final String add = "add";
+    private static final String isSuccessfulOutcome = "isSuccessfulOutcome";
+    private static final String getFailureDescription = "getFailureDescription";
+    private static final String createCompositeOperation = "createCompositeOperation";
+    private static final String getProcessState = "getProcessState";
+    private static final String close = "close";
+    private static final String stop = "stop";
+    private static final String asString = "asString";
+    private static final String READ_ATTRIBUTE_OPERATION = "READ_ATTRIBUTE_OPERATION";
+    private static final String NAME = "NAME";
+    private static final String OP = "OP";
+    private static final String RESULT = "RESULT";
+
+    private static Class<?> EmbeddedManagedProcessClass;
+    private static Class<?> EmbeddedProcessFactoryClass;
+    private static Class<?> ModelControllerClientClass;
+    private static Class<?> ModelNodeClass;
+    private static Class<?> OperationsClass;
+    private static Class<?> ClientConstantsClass;
+
+    private static Method createHostControllerMethod;
+    private static Method createStandaloneServerMethod;
+    private static Method startMethod;
+    private static Method getModelControllerClientMethod;
+    private static Method executeMethod;
+    private static Method fromJSONStringMethod;
+    private static Method getMethod;
+    private static Method setMethod;
+    private static Method addMethod;
+    private static Method isSuccessfulOutcomeMethod;
+    private static Method getFailureDescriptionMethod;
+    private static Method createCompositeOperationMethod;
+    private static Method getProcessStateMethod;
+    private static Method closeMethod;
+    private static Method stopMethod;
+    private static Method asStringMethod;
+
+    static String READ_ATTRIBUTE_OPERATION_FIELD_VALUE;
+    static String NAME_FIELD_VALUE;
+    static String OP_FIELD_VALUE;
+    static String RESULT_FIELD_VALUE;
+
+    private static ServerBridge INSTANCE;
+
+    private ServerBridge() {
+    }
+
+    static ServerBridge get(ClassLoader cl) throws ConfigGeneratorException {
+        if (INSTANCE == null) {
+            initializeEmbedded(cl);
+            INSTANCE = new ServerBridge();
+        }
+        return INSTANCE;
```

Comment:
Are we concerned about thread-safety here? It could potentially initialize more than one instance. I assume too all the instances being static is okay because this should only be used once per-class loader. IoW the `ServerBridge` is only used once. If not, that could be an issue if reusing the same class loader.

Reply:
Done, Thanks.